coriolis_openstack_utils/actions.py

```diff
@@ -1090,3 +1091,143 @@ def execute_operations(self):
             'dest_network_name': dest_network_name}
 
         return dest_subnet
+
+
+class NetworkCreationAction(Action):
+    action_type = ACTION_TYPE_CHECK_CREATE_NETWORK
+    NEW_NETWORK_DESCRIPTION = (
+        "Created by the Coriolis OpenStack utilities for source subnet '%s'.")
+
+    def __init__(self, source_client, destination_client, action_payload):
+        """
+        param action_payload: dict(): payload (params) for the action
+        must contain 'src_network_id'
+                     'dest_tenant_id'
+        """
+        # NOTE no openstack_client or coriolis_client
+
+        super(NetworkCreationAction, self).__init__(
+            None, None, action_payload)
+        self._source_client = source_client
+        self._destination_client = destination_client
+
+    @property
+    def network_name_format(self):
+        return CONF.destination.new_network_name_format
+
+    def check_already_done(self):
+        src_network = networks.get_network(
+            self._source_client, self.payload['src_network_id'])
+
+        dest_network_name = self.get_new_network_name()
+        conflicting = networks.list_networks(
+            self._destination_client, self.payload['dest_tenant_id'],
+            filters={'name': dest_network_name})
+
+        for dest_network in conflicting:
+            if networks.check_network_similarity(
+                    src_network, dest_network,
+                    self._source_client, self._destination_client):
+                LOG.info("Found destination network '%s' with same "
+                         "information and subnets as source network '%s'."
+                         % (dest_network_name, src_network['name']))
+                return {"done": True, "result": dest_network['id']}
+
+        if len(conflicting) == 1:
+            raise Exception("Found destination network with "
+                            "with name '%s' but different attributes!"
+                            % dest_network_name)
+        elif conflicting:
+            raise Exception("Found multiple destination networks with "
+                            "with name '%s'! Aborting subnet "
+                            "migration." % dest_network_name)
+
+        return {"done": False, "result": None}
+
+    def equivalent_to(self, other_action):
+        if other_action.action_type == self.action_type:
+            if self.payload['src_network_id'] == (
+                    other_action.payload.get('src_network_id')):
+                return (self.payload.get('dest_tenant_id') ==
```

Comment:
Please try to avoid writing logic in a `return`. Just append this check to the `if`'s conditional and just do a `return True` at the end.